shm/Map.cpp

```diff
@@ -1,29 +1,28 @@
 #include "Map.hpp"
 #include <algorithm>
 #include <random>
-#include <iostream>
+#include <cstddef>
 
 Map::Map(){
-    islandsLocations_.reserve(amountIslands);
-    generateIslands();
+    islandsLocations_.reserve(amountOfIslands);
+    void generateIslands(int numIslandsToGenerate);
 }
 
-void Map::generateIslands() {
+void Map::generateIslands(int numIslandsToGenerate) {
     std::random_device rd;
     std::mt19937 gen(rd());
-    std::uniform_int_distribution<> distrib(widthIsland, heightIsland);
+    std::uniform_int_distribution<> distrib(minPositionXY, maxPositionXY);
     
-      for (size_t i = 0; i < amountIslands; i++) { 
-        size_t positionX; 
-        size_t positionY; 
-        do { 
-            size_t positionX = distrib(gen); 
-            size_t positionY = distrib(gen);
-        } while (std::any_of(islandsLocations_.begin(), islandsLocations_.end(), [positionX, positionY](auto& island) { 
-            return island.getPosition() == Coordinates(positionX, positionY); 
-        })); 
-        islandsLocations_.push_back(Island(positionX, positionY)); 
-    }
+    do { 
+        size_t positionX = distrib(gen); 
+        size_t positionY = distrib(gen); 
+        if (std::any_of(islandsLocations_.begin(), islandsLocations_.end(), 
+                [positionX, positionY](auto& island) { 
+                return island.getPosition() == Coordinates(positionX, positionY); 
+            })) { 
+            islandsLocations_.push_back(Island(positionX, positionY));
```

Comment:
As said earlier this can be replaced by 
```suggestion
            addIsland({positionX, positionY});
```

Not a big deal if you want to stick with your solution.